moto/route53/models.py

```diff
@@ -277,18 +277,15 @@ def get_all_hosted_zones(self):
         return self.zones.values()
 
     def get_hosted_zone(self, id_):
-        return self.zones.get(id_)
+        return self.zones.get(id_.lstrip("/hostedzone/"))
```

Comment:
It looks like `lstrip()` doesn't behave like this. See [here](https://docs.python.org/2/library/string.html#string.lstrip).

In the cases where `id_` starts with one of the letters in `/hostedzone/`, it is getting removed. For example, `'e67ec7b9'.lstrip("/hostedzone/")` ->`'67ec7b9'`

One option is to change to something like `id_.lstrip("/hostedzone")[0]`.

FWIW, I've been bitten by the unintuitiveness of `lstrip()` in the past many times as well.